### ISSUE TITLE
Improved samples

### ISSUE THREAD

**Issue by Joelius300** [mariusmuntean/ChartJs.Blazor#122]
Currently our library has three sample projects. That's too much.  
Our samples also don't show nearly all features Chart.js resp. our library has.  
Nor are they analogous to the [official Chart.js samples](https://www.chartjs.org/samples/latest/) which [generates confusion](https://github.com/mariusmuntean/ChartJs.Blazor/issues/111).

### Client- / Server-side
Currently we have lots of redundancies because of our three projects. We should eliminate two of them and run them in dual mode. There's already an [amazing resource for running blazor in dual mode](https://github.com/Suchiman/BlazorDualMode) which we can take as reference. Once this is done, we can also update https://www.iheartblazor.com with these samples so you can swich between client and server without having to build the samples yourself.

### Porting samples
We should try to mirror the offical samples in order to show our consumers how to achieve the same thing. This has the many benefits like more clarity for the simple user when trying to achieve something and a reference for the people who already know Chart.js. It's also simpler for us to maintain and to give support.  
Of course some additional samples might be needed e.g. for the callback stuff, especially when you consider the [server-side incompatibility](https://github.com/mariusmuntean/ChartJs.Blazor/issues/90).

Below you can find a list of samples we should port.
##### Bar charts
- [ ] [Vertical](https://www.chartjs.org/samples/latest/charts/bar/vertical.html)
- [ ] [Horizontal](https://www.chartjs.org/samples/latest/charts/bar/horizontal.html)
- [ ] [Stacked](https://www.chartjs.org/samples/latest/charts/bar/stacked.html)
- [ ] [Floating](https://www.chartjs.org/samples/latest/charts/bar/float.html)

##### Line charts
- [ ] [Basic](https://www.chartjs.org/samples/latest/charts/line/basic.html)
- [ ] [Stepped](https://www.chartjs.org/samples/latest/charts/line/stepped.html)
- [ ] [Interpolation](https://www.chartjs.org/samples/latest/charts/line/interpolation-modes.html)
- [ ] [Line styles](https://www.chartjs.org/samples/latest/charts/line/line-styles.html)
- [ ] [Point styles](https://www.chartjs.org/samples/latest/charts/line/point-styles.html)
- [ ] [Point sizes](https://www.chartjs.org/samples/latest/charts/line/point-sizes.html)

##### Area charts
- [ ] [Boundaries (line)](https://www.chartjs.org/samples/latest/charts/area/line-boundaries.html)
- [ ] [Datasets (line)](https://www.chartjs.org/samples/latest/charts/area/line-datasets.html)
- [ ] [Stacked (line)](https://www.chartjs.org/samples/latest/charts/area/line-stacked.html)
- [ ] [Radar](https://www.chartjs.org/samples/latest/charts/area/radar.html)

##### Other charts
- [ ] [Bubble](https://www.chartjs.org/samples/latest/charts/bubble.html)
- [ ] [Scatter](https://www.chartjs.org/samples/latest/charts/scatter/basic.html)
- [ ] [Doughnut](https://www.chartjs.org/samples/latest/charts/doughnut.html)
- [ ] [Pie](https://www.chartjs.org/samples/latest/charts/pie.html)
- [ ] [Polar area](https://www.chartjs.org/samples/latest/charts/polar-area.html)
- [ ] [Radar](https://www.chartjs.org/samples/latest/charts/radar.html)
- [ ] [Combo bar/line](https://www.chartjs.org/samples/latest/charts/combo-bar-line.html)

##### Linear scale
- [ ] [Step size](https://www.chartjs.org/samples/latest/scales/linear/step-size.html)
- [ ] [Min & max](https://www.chartjs.org/samples/latest/scales/linear/min-max.html)
- [ ] [Min & max (suggested)](https://www.chartjs.org/samples/latest/scales/linear/min-max-suggested.html)

##### Logarithmic scale
- [ ] [Line](https://www.chartjs.org/samples/latest/scales/logarithmic/line.html)
- [ ] [Scatter](https://www.chartjs.org/samples/latest/scales/logarithmic/scatter.html)

##### Time scale
- [ ] [Line](https://www.chartjs.org/samples/latest/scales/time/line.html)
- [ ] [Line (point data)](https://www.chartjs.org/samples/latest/scales/time/line-point-data.html)
- [ ] [Combo](https://www.chartjs.org/samples/latest/scales/time/combo.html)

##### Scale options
- [ ] [Grid lines display](https://www.chartjs.org/samples/latest/scales/gridlines-display.html)
- [ ] [Grid lines style](https://www.chartjs.org/samples/latest/scales/gridlines-style.html)
- [ ] [Non numeric Y Axis](https://www.chartjs.org/samples/latest/scales/non-numeric-y.html)
- [ ] [Axes Labels](https://www.chartjs.org/samples/latest/scales/axes-labels.html)
- [ ] [Center Positioning](https://www.chartjs.org/samples/latest/scales/axis-center-position.html)

##### Legend
- [ ] [Positioning](https://www.chartjs.org/samples/latest/legend/positioning.html)
- [ ] [Legend Title](https://www.chartjs.org/samples/latest/legend/title.html)
- [ ] [Point style](https://www.chartjs.org/samples/latest/legend/point-style.html)
- [ ] [Callbacks](https://www.chartjs.org/samples/latest/legend/callbacks.html)

##### Title
- [ ] [Alignment](https://www.chartjs.org/samples/latest/title/alignment.html)

##### Tooltip
- [ ] [Positioning](https://www.chartjs.org/samples/latest/tooltips/positioning.html)
- [ ] [Interactions](https://www.chartjs.org/samples/latest/tooltips/interactions.html)
- [ ] [Callbacks](https://www.chartjs.org/samples/latest/tooltips/callbacks.html)
- [ ] [Border](https://www.chartjs.org/samples/latest/tooltips/border.html)

##### Bar charts
- [ ] [Multi axis](https://www.chartjs.org/samples/latest/charts/bar/multi-axis.html)
- [ ] [Stacked groups](https://www.chartjs.org/samples/latest/charts/bar/stacked-group.html)

##### Line charts
- [ ] [Multi axis](https://www.chartjs.org/samples/latest/charts/line/multi-axis.html)

##### Other charts
- [ ] [Scatter - Multi axis](https://www.chartjs.org/samples/latest/charts/scatter/multi-axis.html)
- [ ] [Multi Series Pie](https://www.chartjs.org/samples/latest/charts/multi-series-pie.html)

##### Time scale
- [ ] [Line (break on 2 day gap)](https://www.chartjs.org/samples/latest/scales/time/line-max-span.html)

##### Scale options
- [ ] [Scriptable Grid lines](https://www.chartjs.org/samples/latest/scales/gridlines-scriptable.html)
- [ ] [Multiline labels](https://www.chartjs.org/samples/latest/scales/multiline-labels.html)
- [ ] [Filtering Labels](https://www.chartjs.org/samples/latest/scales/filtering-labels.html)
- [ ] [Toggle Scale Type](https://www.chartjs.org/samples/latest/scales/toggle-scale-type.html)
- [ ] [Custom major ticks](https://www.chartjs.org/samples/latest/scales/financial.html)

##### Tooltip
- [ ] [HTML tooltips (line)](https://www.chartjs.org/samples/latest/tooltips/custom-line.html)
- [ ] [HTML tooltips (pie)](https://www.chartjs.org/samples/latest/tooltips/custom-pie.html)
- [ ] [HTML tooltips (points)](https://www.chartjs.org/samples/latest/tooltips/custom-points.html)

##### Scriptable
- [ ] [Bar Chart](https://www.chartjs.org/samples/latest/scriptable/bar.html)
- [ ] [Bubble Chart](https://www.chartjs.org/samples/latest/scriptable/bubble.html)
- [ ] [Pie Chart](https://www.chartjs.org/samples/latest/scriptable/pie.html)
- [ ] [Line Chart](https://www.chartjs.org/samples/latest/scriptable/line.html)
- [ ] [Polar Area Chart](https://www.chartjs.org/samples/latest/scriptable/polar.html)
- [ ] [Radar Chart](https://www.chartjs.org/samples/latest/scriptable/radar.html)

##### Animations
- [ ] [Delay](https://www.chartjs.org/samples/latest/animations/delay.html)
- [ ] [Drop](https://www.chartjs.org/samples/latest/animations/drop.html)
- [ ] [Loop](https://www.chartjs.org/samples/latest/animations/loop.html)

##### Advanced
- [ ] [Progress bar](https://www.chartjs.org/samples/latest/advanced/progress-bar.html)
- [ ] [Polar Area Radial Gradient](https://www.chartjs.org/samples/latest/advanced/radial-gradient.html)
- [ ] [Line Gradient](https://www.chartjs.org/samples/latest/advanced/line-gradient.html)

**Comment by mariusmuntean** [comment]
I agree with improving the samples; that should make it much easier to understand how to achieve a certain result with ChartJs.Blazor.

**Comment by Joelius300** [comment]
@mariusmuntean What do you think of my priority grouping (important / additional)?
